code/modules/shuttle/shuttle.dm

```diff
@@ -196,6 +196,30 @@
 	if(roundstart_template)
 		SSshuttle.manipulator.action_load(roundstart_template, src)
 
+	var/obj/docking_port/mobile/port = get_docked()
+	if(!port)
+		if(roundstart_template)
+			CRASH("No mobile dock found even though a shuttle loaded")
+		return
+
+	var/list/static/shuttle_id = list()
+	var/idnum
+	if(!shuttle_id[roundstart_template])
+		shuttle_id[roundstart_template] = idnum = 1
+	else
+		idnum = shuttle_id[roundstart_template]++
+	if(port.id == initial(port.id))
+		port.id = "[initial(port.id)][idnum]"
+	if(port.name == initial(port.name))
+		port.name = "[initial(port.name)] [idnum]"
```

Comment:
Will this end up naming every single shuttle "NT Recovery Ship 1" etc.?

Reply:
every shuttle? no